### ISSUE TITLE
Question on customQueries

### ISSUE THREAD

**Issue by developer-rakeshpaul** [RisingStack/graffiti-mongoose#145]
## Overview of the Issue

When we define customQueries as functions, the value passed to the function has only graphql InputType objects. I am curious to know why it is done that way.

In my current project i am trying to utilize the customQueries like below 

```javascript
// MY Custom query definition
import {
  GraphQLList,
  GraphQLString,
  GraphQLNonNull,
} from 'graphql';

import {
  getProjection,
} from '../../projections';

import Algorithm from '../../schema/algorithm';

const getAlgorithmsByType = (types) => ({
  getAlgorithmsByType: {
    type: new GraphQLList(types.Algorithm),
    args: {
      type: {
        name: 'type',
        type: new GraphQLNonNull(GraphQLString),
      },
    },
    resolve(root, { type }, source, { fieldASTs }) {
      const projections = getProjection(fieldASTs);
      return Algorithm .find( { type }, projections);
    },
  },
});

export {
  getAlgorithmsByType,
};

```

```javascript
//My graphql schema generation code
import {
  getCustomQueries,
} from './custom';

const models = [];
const schemas = requireDir('./schema');
Object.keys(schemas).map((key) => {
  const schema = schemas[key].default;
  models.push(schema);
});

const customQueries = getCustomQueries();

const options = {
  mutation: true, // mutation fields can be disabled
  allowMongoIDMutation: false, // mutation of mongo _id can be enabled
  hooks,
  customQueries,
};

const schema = getSchema(models, options);
export default schema;

```

For the above code to work the customQuery method should get the model type as an argument to the getCustomQueries

I was able to get it working by changing the line number 269 in the schema\schema.js file
```javascript
      ? customQueries(mapValues(types, (type) => createInputObject(type))) 
 ```
to
```javascript
      ? customQueries(types) 
 ```

It would be great to know if there are any alternate ways to handle my requirement.

**Comment by tothandras** [comment]
Hi @developer-rakeshpaul, you get a `GraphQLInputObjectType` in your customQueries function as the first parameter, not the actual types with `Algorithm`. Here is a PR that can help you without breaking change: #148. Let me know if it helps, I can merge and publish it then.

**Comment by yoadsn** [comment]
Although one has to ask how common is it to use the input types for Queries, where inputs are most likely simple types and not complex like with mutations..

**Comment by tothandras** [comment]
I agree, but I am not sure if anyone uses that. We can have a breaking
change if it's more reasonable.

**Comment by yoadsn** [comment]
It is unfortunate but I agree with you - it is probably best not to break comp. and just add another arg.

**Comment by tothandras** [comment]
okay, I will merge the PR as it is and release it

**Comment by tothandras** [comment]
released in version `5.3.0` 😉

**Comment by developer-rakeshpaul** [comment]
@tothandras thanks for the PR..
